airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTest.java

```diff
@@ -108,106 +110,128 @@ class BigQueryDestinationTest {
 
-  protected JsonNode config;
+  protected static String projectId;
+  protected static String datasetId;
+  protected static JsonNode config;
+  protected static JsonNode configWithProjectId;
+  protected static JsonNode configWithBadProjectId;
+  protected static JsonNode insufficientRoleConfig;
+  protected static JsonNode nonBillableConfig;
+  protected static JsonNode gcsStagingConfig; //default BigQuery config. Also used for setup/teardown
   protected BigQuery bigquery;
   protected Dataset dataset;
-  protected ConfiguredAirbyteCatalog catalog;
-  protected boolean tornDown = true;
+  protected static Map<String, JsonNode> configs;
+  protected static ConfiguredAirbyteCatalog catalog;
+
+  private AmazonS3 s3Client;
 
-  private static Stream<Arguments> datasetIdResetterProvider() {
-    // parameterized test with two dataset-id patterns: `dataset_id` and `project-id:dataset_id`
+  private Stream<Arguments> successTestConfigProvider() {
     return Stream.of(
-        Arguments.arguments(new DatasetIdResetter(config -> {})),
-        Arguments.arguments(new DatasetIdResetter(
-            config -> {
-              final String projectId = config.get(BigQueryConsts.CONFIG_PROJECT_ID).asText();
-              final String datasetId = config.get(BigQueryConsts.CONFIG_DATASET_ID).asText();
-              ((ObjectNode) config).put(BigQueryConsts.CONFIG_DATASET_ID,
-                  String.format("%s:%s", projectId, datasetId));
-            })));
+        Arguments.of("config"),
+        Arguments.of("configWithProjectId"),
+        Arguments.of("gcsStagingConfig")
+    );
   }
 
-  @BeforeEach
-  void setup(final TestInfo info) throws IOException {
-    if (info.getDisplayName().equals("testSpec()")) {
-      return;
-    }
+  private Stream<Arguments> failCheckTestConfigProvider() {
+    return Stream.of(
+        Arguments.of("configWithBadProjectId", "User does not have bigquery.datasets.create permission in project"),
+        Arguments.of("insufficientRoleConfig", "User does not have bigquery.datasets.create permission"),
+        Arguments.of("nonBillableConfig", "Access Denied: BigQuery BigQuery: Streaming insert is not allowed in the free tier")
+    );
+  }
 
-    if (!Files.exists(CREDENTIALS_PATH)) {
-      throw new IllegalStateException(
-          "Must provide path to a big query credentials file. By default {module-root}/config/credentials.json. Override by setting setting path with the CREDENTIALS_PATH constant.");
+  private Stream<Arguments> failWriteTestConfigProvider() {
+    return Stream.of(
+        Arguments.of("configWithBadProjectId", "User does not have bigquery.datasets.create permission in project"),
+        Arguments.of("insufficientRoleConfig", "Permission bigquery.tables.create denied")
+    );
+  }
+
+  @BeforeAll
+  public static void beforeAll() throws IOException {
+    for(Path path : ALL_PATHS) {
+      if (!Files.exists(path)) {
+        throw new IllegalStateException(
```

Comment:
Small thing - I believe this exception will be thrown on the first file to not exist, then exit the loop. It might be more convenient if all missing files are shown in the error message.

```
final List<Path> missingPaths = ALL_PATHS.stream().filter(path -> !Files.exist(path)).collect(Collectors.toList());
if (!missingPaths.isEmpty()) {
  throw new IllegalStateException(
  // ...
  )
}

```